torchao/quantization/subclass.py

```diff
@@ -805,18 +861,18 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
                     # TODO: enable mps path as well
                     # per channel int8 weight only quantizated mm
                     return torch.ops.aten._weight_int8pack_mm(input_tensor.contiguous(), weight_qtensor.int_data, weight_qtensor.scale)
+                else:
+                    weight_tensor = weight_qtensor.dequantize()
+                    return torch.nn.functional.linear(input_tensor, weight_tensor, bias)
             else:
-                # dynamic quantization
-                input_tensor = weight_qtensor.input_quant_func(input_tensor)
-                input_tensor = input_tensor.dequantize()
-            weight_tensor = weight_qtensor.dequantize()
-            return torch.nn.functional.linear(input_tensor, weight_tensor, bias)
-
+                if isinstance(input_tensor, AffineQuantizedTensor):
+                    input_tensor = input_tensor.dequantize()
+                return torch.nn.functional.linear(input_tensor, weight_tensor, bias)
         try:
             with torch._C.DisableTorchFunctionSubclass():
                 return func(*args, **kwargs)
         except:
-            print(f"ERR: subclass doesn't implement {func}")
+            print(f"ERR: AffineQuantizedTensor subclass doesn't implement {func}")
```

Comment:
A blanked `except` here will say that `func` is not implemented when in fact `func` might have raised a valid `ValueError` or such when an argument was chosen incorrectly. This error should also now be raised from within torch dispatch itself

Reply:
Sure I can change the error message. 
you mean we want to reraise the error here?

Reply:
followed up offline, will remove try except here